src/workerd/api/cache.c++

```diff
@@ -516,17 +520,29 @@ jsg::Promise<bool> Cache::delete_(
   });
 }
 
-kj::Own<kj::HttpClient> Cache::getHttpClient(
-    IoContext& context, kj::Maybe<kj::String> cfBlobJson, kj::LiteralStringConst operationName) {
+kj::Own<kj::HttpClient> Cache::getHttpClient(IoContext& context,
+    kj::Maybe<kj::String> cfBlobJson,
+    kj::LiteralStringConst operationName,
+    kj::StringPtr url,
+    kj::Maybe<jsg::ByteString> cacheControl) {
   auto span = context.makeTraceSpan(operationName);
   auto userSpan = context.makeUserTraceSpan(operationName);
 
+  userSpan.setTag("url.full"_kjc, kj::str(url));
+  // TODO(o11y): We add a tag for must-revalidate as a proof-of-concept for now (see
+  // https://developers.cloudflare.com/cache/concepts/cache-control/#revalidation). Parse
+  // cacheControl to support other revalidation, cacheability and expiration directives too.
+  KJ_IF_SOME(c, cacheControl) {
+    if (c == "must-revalidate") {
+      userSpan.setTag("cache_control.revalidation"_kjc, kj::str("must-revalidate"));
+    }
+  }
   auto cacheClient = context.getCacheClient();
   auto httpClient = cacheName
                         .map([&](kj::String& n) {
     return cacheClient->getNamespace(n, kj::mv(cfBlobJson), span);
```

Comment:
Perhaps these should also need to be updated to propagate the user span, but since we attach the user span to httpClient below we should be fine in terms of span lifetime which is what we need for now.